walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/data/Task.java

```diff
@@ -26,4 +26,347 @@ public Task(long id, String title, long timestamp) {
+      return;
+    }
+
+    int hintNumber = (int) hintEntity.getProperty("hintNumber");
+    String text = (String) hintEntity.getProperty("key");
+    double latitude = (double) hintEntity.getProperty("latitude");
+    double longitude = (double) hintEntity.getProperty("longitude");
+    Coordinates startingLocation;
+    startingLocation = new Coordinates(latitude, longitude);
+
+    Hint.Builder hint = new Hint.Builder(hint.getHintID, hintNumber);
+    hint.setLocation(startingLocation).setText(text);
+    return hint.build();
+  }
+
+  /**
+    * Stores the static data of a single user in datastore as an entity
+    * @param user a User variable representing a single instance of a user.
+  */
+  public void storeUser(User user) {
+    Entity userEntity = new Entity("User", user.getUserID());
+    userEntity.setProperty("username", user.getUsername());
+    userEntity.setProperty("firstname", user.getFirstName());
+    userEntity.setProperty("lastname", user.getLastName());
+    userEntity.setProperty("gamesCreated", user.getGamesCreated());
+    userEntity.setProperty("profilePictureUrl", user.getProfilePictureUrl());
+    userEntity.setProperty("numGamesFinished", user.getNumGamesFinished());
+    String numbersJson = gson.toJson(user.getGamesCompletedWithTime());
+    userEntity.setProperty("gamesCompletedWithTime", numbersJson);
+    userEntity.setProperty("numGamesCreated", user.getNumGamesCreated());
+    datastore.put(userEntity);
+  }
+  
+  /**
+    * Updates an entity of a single user data in datastore
+    * @param user a User variable representing a single instance of a user.
+  */
+  public void updateUser(User user) {
+    Key userEntityKey = KeyFactory.createKey("User", user.getUserID());
+    Entity userEntity;
+    try {
+      userEntity = datastore.get(userEntityKey);
+    }
+    catch(Exception e) {
+      return;
+    }
+      
+    userEntity.setProperty("username", user.getUsername());
+    userEntity.setProperty("firstname", user.getFirstName());
+    userEntity.setProperty("lastname", user.getLastName());
+    userEntity.setProperty("gamesCreated", user.getGamesCreated());
+    userEntity.setProperty("profilePictureUrl", user.getProfilePictureUrl());
+    userEntity.setProperty("numGamesFinished", user.getNumGamesFinished());
+    String numbersJson = gson.toJson(user.getGamesCompletedWithTime());
+    userEntity.setProperty("gamesCompletedWithTime", numbersJson);
+    userEntity.setProperty("numGamesCreated", user.getNumGamesCreated());
+    datastore.put(userEntity);
+  }
+  
+  /**
+    * Retrieves a single user entity from the datastore.
+    * @param userID the unique code used to identify this specific user.
+    * @return a user object with the properties specified in the Builder.
+  */
+  public Game retrieveUser(String userID) {
+    Key userEntityKey = KeyFactory.createKey("User", userID);
+    Entity userEntity;
+    try {
+      userEntity = datastore.get(userEntityKey);
+    }
+    catch(Exception e) {
```

Comment:
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/DatastoreService.html#get-com.google.appengine.api.datastore.Key-

The DatastoreService#get(Key) method will throw an EntityNotFoundException. So you can re-throw the exception like below 

public Game retrieveUser(String userID) throws EntityNotFoundException {
   Key userEntityKey = KeyFactory.createKey("User", userID);
    Entity userEntity = datastore.get(userEntityKey);
    .....
}

In this case, the caller of this method should capture the EntityNotFoundException

Reply:
sorry this piece of code wasn't meant to be included